SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -16,16 +16,115 @@
 
 DEFINE_LOG_CATEGORY(LogInterestFactory);
 
-namespace
+namespace SpatialGDK
+{
+// The checkout radius constraint is built once for all actors in CreateCheckoutRadiusConstraint as it is equivalent for all actors.
+// It is built once per net driver initialisation.
+static QueryConstraint ClientCheckoutRadiusConstraint;
+
+InterestFactory::InterestFactory(AActor* InActor, const FClassInfo& InInfo, USpatialClassInfoManager* InClassInfoManager, USpatialPackageMapClient* InPackageMap)
+	: Actor(InActor)
+	, Info(InInfo)
+	, ClassInfoManager(InClassInfoManager)
+	, PackageMap(InPackageMap)
 {
-static TMap<UClass*, float> ClientInterestDistancesSquared;
 }
 
-namespace SpatialGDK
+void InterestFactory::CreateClientCheckoutRadiusConstraint(USpatialClassInfoManager* ClassInfoManager)
+{
+	// Checkout Radius constraints are defined by the NetCullDistanceSquared property on actors.
+	//   - Checkout radius is a RelativeCylinder constraint on the player controller.
+	//   - NetCullDistanceSquared on AActor is used to define the default checkout radius with no other constraints.
+	//   - NetCullDistanceSquared on other actor types is used to define additional constraints if needed.
+	//   - If a subtype defines a radius smaller than a parent type, then its requirements are already captured.
+	//   - If a subtype defines a radius larger than all parent types, then it needs an additional constraint.
+	//   - Other than the default from AActor, all radius constraints also include Component constraints to
+	//     capture specific types, including all derived types of that actor.
+
+	QueryConstraint CheckoutRadiusConstraint;
```

Comment:
Just operate on the `ClientCheckoutRadiusConstraint ` directly?

Reply:
it was written this way in order to be able to extract it, and assign `ClientCheckoutRadiusConstraint` to a function call that can be tested, but need to work out how to test given this function depends on implicit state (maybe the state in testgym is enough?)